### PR TITLE
DTS-3385: upgrade hadoop version

### PR DESCRIPTION
**Pull request by pranavsuku-db**
upgrading hadoop version so it is compatible with delta-kernel

### CODE REVIEW

**Review by linzhou-db — APPROVED** [review]

**Comment by linzhou-db** [comment]
could you rerun the tests, and run them locally to double check?

File: build.sbt [hunk 1]
```diff
@@ -145,22 +145,37 @@ lazy val server = (project in file("server")) enablePlugins(JavaAppPackaging) se
...
       ExclusionRule("com.google.guava", "guava")
     ),
-    "org.apache.parquet" % "parquet-hadoop" % "1.10.1" excludeAll(
+    "org.apache.parquet" % "parquet-hadoop" % "1.12.3" excludeAll(
       ExclusionRule("com.fasterxml.jackson.core"),
       ExclusionRule("com.fasterxml.jackson.module"),
       ExclusionRule("com.google.guava", "guava")
     ),
-    "io.delta" %% "delta-standalone" % "0.5.0" excludeAll(
+    "io.delta" % "delta-kernel-api" % "3.2.0" excludeAll(
```
Comment (linzhou-db) [inline build.sbt:163]: we don't need to include them twice?

File: build.sbt [hunk 2]
```diff
@@ -145,22 +145,37 @@ lazy val server = (project in file("server")) enablePlugins(JavaAppPackaging) se
...
+      ExclusionRule("com.fasterxml.jackson.core"),
+      ExclusionRule("com.fasterxml.jackson.module"),
+      ExclusionRule("com.google.guava", "guava")
+    ),
+    "io.delta" % "delta-kernel-defaults" % "3.2.0" excludeAll(
+      ExclusionRule("com.fasterxml.jackson.core"),
+      ExclusionRule("com.fasterxml.jackson.module"),
+      ExclusionRule("com.google.guava", "guava")
+    ),
+    "io.delta" % "delta-storage" % "3.2.0" excludeAll(
```
Comment (linzhou-db) [inline build.sbt:173]: and exclude this delta-storage library?

File: build.sbt (hunk 1 above)
Comment (pranavsuku-db) [reply build.sbt]: fixed

File: build.sbt (hunk 2 above)
Comment (pranavsuku-db) [reply build.sbt]: fixed

**Review by linzhou-db — APPROVED** [review]
LGTM, let's debug the other test failure separately.

File: build.sbt [hunk 3]
```diff
@@ -145,22 +145,27 @@ lazy val server = (project in file("server")) enablePlugins(JavaAppPackaging) se
...
       ExclusionRule("com.google.guava", "guava")
     ),
-    "org.apache.parquet" % "parquet-hadoop" % "1.10.1" excludeAll(
+    "org.apache.parquet" % "parquet-hadoop" % "1.12.3" excludeAll(
       ExclusionRule("com.fasterxml.jackson.core"),
       ExclusionRule("com.fasterxml.jackson.module"),
       ExclusionRule("com.google.guava", "guava")
     ),
-    "io.delta" %% "delta-standalone" % "0.5.0" excludeAll(
+    "io.delta" % "delta-kernel-defaults" % "3.2.0" excludeAll(
```
Comment (linzhou-db) [inline build.sbt:163]: This one is also a duplicate of the import below?

Comment (pranavsuku-db) [reply build.sbt:163]: fixed

File: client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala [hunk 4]
```diff
@@ -114,7 +115,7 @@ private[sharing] class RandomAccessHttpInputStream(
   }
 
   private def createHttpRequest(start: Long): HttpRequestBase = {
-    val request = new HttpGet(uri)
+    val request = new HttpGet(URI.create(uri))
```
Comment (linzhou-db) [inline client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala:118]: this is not needed

File: client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala [hunk 5]
```diff
@@ -40,6 +40,7 @@
 package io.delta.sharing.client
 
 import java.io.{EOFException, InputStream, IOException}
+import java.net.URI
```
Comment (linzhou-db) [inline client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala:43]: you can revert this.

File: server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala [hunk 6]
```diff
@@ -175,7 +176,12 @@ object AbfsFileSigner {
     val getRelativePathMethod = classOf[AzureBlobFileSystemStore]
       .getDeclaredMethod("getRelativePath", classOf[Path])
     getRelativePathMethod.setAccessible(true)
-    getRelativePathMethod.invoke(abfsStore, path).asInstanceOf[String]
+    var relativePath = getRelativePathMethod.invoke(abfsStore, path).asInstanceOf[String]
+    // remove duplicate separator character for azure relative path
```
Comment (linzhou-db) [inline server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala:180]: And could you add this link in the comment as a reference of why we are doing this change: https://github.com/apache/hadoop/commit/af98f32f7dbb9d71915690b66f12c33758011450#diff-94925ffd3b21968d7e6b476f7e85f68f5ea326f186262017fad61a5a6a3815cbL1215

**Review by linzhou-db — APPROVED** [review]

File: client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala (hunk 4 above)
Comment (pranavsuku-db) [reply client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala:118]: fixed

File: client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala (hunk 5 above)
Comment (pranavsuku-db) [reply client/src/main/scala/io/delta/sharing/client/RandomAccessHttpInputStream.scala:43]: fixed

File: server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala (hunk 6 above)
Comment (pranavsuku-db) [reply server/src/main/scala/io/delta/sharing/server/CloudFileSigner.scala:181]: done

**Pull request merged**